javascript/src/schema/schema.ts

```diff
@@ -257,13 +258,55 @@ export interface AudioPartDelta {
    */
   id?: string;
 }
+/**
+ * A part of the message that contains reasoning/thinking content.
+ *
+ * This interface was referenced by `LlmSdk`'s JSON-Schema
+ * via the `definition` "ReasoningPart".
+ */
+export interface ReasoningPart {
+  type: "reasoning";
+  /**
+   * The reasoning/thinking content
+   */
+  reasoning: string;
+  /**
+   * Whether this is a summary of the reasoning process
+   */
+  summary?: boolean;
+  /**
+   * The optional ID of the part.
+   */
+  id?: string;
+}
+
+/**
+ * This interface was referenced by `LlmSdk`'s JSON-Schema
+ * via the `definition` "ReasoningPartDelta".
+ */
+export interface ReasoningPartDelta {
+  type: "reasoning";
```

Comment:
just a small suggestion here, does it make sense to have the union type here based on the `type` having an enum too?

`ReasoningTypeEnum`, for example

Reply:
The union is implemented elsewhere

```
export interface ContentDelta {
  index: number;
  part: TextPartDelta | ToolCallPartDelta | AudioPartDelta | ReasoningPartDelta;
}
```

and the type: "reasoning" is following the other PartDeltas above
i dont see a big benefit to changing it!